lib/resty/acme/autossl.lua

```diff
@@ -510,25 +512,38 @@ function AUTOSSL.ssl_certificate()
     end
   end
 
+  for i, typ in ipairs(domain_key_types) do
+    get_cert_inline(i, typ, AUTOSSL.config.blocking)
+  end
+
   if domain_key_types_count ~= chains_set then
-    ngx.timer.at(0, function()
-      for i, typ in ipairs(domain_key_types) do
-        if not chains_set[i] then
-          local err = AUTOSSL.update_cert({
-            domain = domain,
-            renew = false,
-            tries = 0,
-            type = typ,
-          })
-
-          if err then
-            log(ngx_ERR, "failed to create ", typ, " certificate for domain ", domain, ": ", err)
-          end
+    local update_cert_inline = function(i, typ)
+      if not chains_set[i] then
+        local err = AUTOSSL.update_cert({
+          domain = domain,
+          renew = false,
+          tries = 0,
+          type = typ,
+        })
+
+        if err then
+          log(ngx_ERR, "failed to create ", typ, " certificate for domain ", domain, ": ", err)
         end
       end
-    end)
-    -- serve fallback cert this time
-    return
+    end
+
+    for i, typ in ipairs(domain_key_types) do
+      if AUTOSSL.config.blocking then
+        -- blocking update_cert and attempt to set cert again
+        update_cert_inline(i, typ)
+        get_cert_inline(i, typ, false)
+      else
+        -- non blocking update_cert, let it fallback to default cert
+        ngx.timer.at(0, function()
```

Comment:
this implies we will create two types of certs at same type, which could be a problem

Reply:
Yes that could happen, but isn't that what the code currently does already? https://github.com/fffonion/lua-resty-acme/blob/15a09bd353545aee61ea3b5a641629db027ed1f0/lib/resty/acme/autossl.lua#L515-L522

However I realize that I moved the `if not chains_set[i] then` condition in the new inline function which isn't great. I just moved it back inside the loop.

Reply:
actually `AUTOSSL.update_cert` is "blocking", so the old behaviour is starting a timer and create two types of certs
sequentially (if configured)

Reply:
I thought the old behavior was starting 2 timers each proceeding to generating certs in parallel (but again I'm not familiar with Resty nginx functions). Whereas the new code creates them sequentially (since it's blocking). But yeah in any case `update_cert` is called for each type.

What issue(s) do you see arising from the blocking version?

Reply:
> I thought the old behavior was starting 2 timers each proceeding to generating certs in parallel 

we only have 1 timer previously 😄 it's outside of the for loop. So if blocking=false, we should keep this behaviour,
otherwise we will end up running 2 timers concurrently. In your code for loop is outside of ngx.timer.at.

Reply:
Oh ok! Thanks for clarifying, I'll try to fix that. :)

Reply:
I pushed a commit which should revert the non blocking mode to its normal behavior.